gst/nnstreamer/elements/gsttensor_trainer.c

```diff
@@ -0,0 +1,1182 @@
+  g_object_class_install_property (gobject_class, PROP_FRAMEWORK,
+      g_param_spec_string ("framework", "Framework", "Neural network framework",
+          DEFAULT_PROP_FRAMEWORK,
+          G_PARAM_READWRITE | GST_PARAM_MUTABLE_READY |
+          G_PARAM_STATIC_STRINGS));
+
+  g_object_class_install_property (gobject_class, PROP_MODEL_CONFIG,
+      g_param_spec_string ("model-config", "Model configuration file path",
+          "Configuration file path for creating model",
+          DEFAULT_STR_PROP_VALUE,
+          G_PARAM_READWRITE | GST_PARAM_MUTABLE_READY |
+          G_PARAM_STATIC_STRINGS));
+
+  g_object_class_install_property (gobject_class, PROP_MODEL_SAVE_PATH,
+      g_param_spec_string ("model-save-path", "Model save path",
+          "File path to save the model", DEFAULT_STR_PROP_VALUE,
+          G_PARAM_READWRITE | GST_PARAM_MUTABLE_READY |
+          G_PARAM_STATIC_STRINGS));
+
+  g_object_class_install_property (gobject_class, PROP_INPUT,
+      g_param_spec_string ("input", "Input dimension",
+          "Input tensors dimension from inner array, up to 4 dimensions ?",
+          DEFAULT_STR_PROP_VALUE,
+          G_PARAM_READWRITE | GST_PARAM_MUTABLE_READY |
+          G_PARAM_STATIC_STRINGS));
+
+  g_object_class_install_property (gobject_class, PROP_INPUTTYPE,
+      g_param_spec_string ("inputtype", "Input tensor element type",
+          "Type of each element of the input tensor ?",
+          DEFAULT_STR_PROP_VALUE,
+          G_PARAM_READWRITE | GST_PARAM_MUTABLE_READY |
+          G_PARAM_STATIC_STRINGS));
+
+  g_object_class_install_property (gobject_class, PROP_PUSH_OUTPUT,
+      g_param_spec_boolean ("push-output", "Push output tensor",
+          "Add output tensors to output GstBuffer", FALSE,
+          G_PARAM_READWRITE | GST_PARAM_MUTABLE_READY |
+          G_PARAM_STATIC_STRINGS));
+
+  g_object_class_install_property (gobject_class, PROP_INPUT_LIST,
+      g_param_spec_uint ("input-lists", "Number of input lists",
+          "Number of input list", 0, NNS_TENSOR_SIZE_LIMIT, 1,
+          G_PARAM_READWRITE | GST_PARAM_MUTABLE_READY |
+          G_PARAM_STATIC_STRINGS));
+
+  g_object_class_install_property (gobject_class, PROP_LABEL_LIST,
+      g_param_spec_uint ("label-lists", "Number of label lists",
+          "Number of label list", 0, NNS_TENSOR_SIZE_LIMIT, 1,
+          G_PARAM_READWRITE | GST_PARAM_MUTABLE_READY |
+          G_PARAM_STATIC_STRINGS));
+
+  g_object_class_install_property (gobject_class, PROP_TRAIN_SAMPLES,
+      g_param_spec_uint ("train-samples", "Number of train samples",
+          "Number of train samples", 0, G_MAXUINT, 1,
+          G_PARAM_READWRITE | GST_PARAM_MUTABLE_READY |
+          G_PARAM_STATIC_STRINGS));
+
+  g_object_class_install_property (gobject_class, PROP_VALID_SAMPLES,
+      g_param_spec_uint ("valid-samples", "Number of valid samples",
+          "Number of valid samples", 0, G_MAXUINT, 1,
+          G_PARAM_READWRITE | GST_PARAM_MUTABLE_READY |
```

Comment:
The decription is not enough.

Reply:
Thank you, I will update another PR.

Reply:
Fixed as much as possible in this PR.
If it's not enough, I'll fix it next time.
Thank you.